bricks/template/hooks/pre_gen.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:mason/mason.dart';
+
+import 'bundles/permission_handler_bundle.dart';
+
+Future<void> run(HookContext context) async {
+  try {
+    final newVars = {};
+    if (context.vars['add_permission_handler'] == true) {
+      newVars.addAll(await addPermissionHandlerVariables());
+    }
+    context.vars = {...context.vars, ...newVars};
+  } catch (e) {
+    context.logger.err(e.toString());
+  }
+}
+
+Future<Map<String, dynamic>> addPermissionHandlerVariables() async {
+  Map<String, dynamic> newVars = {};
+  final generator = await MasonGenerator.fromBundle(permissionHandlerBundle);
+  generator.partials.forEach((filePath, content) {
+    if (filePath.startsWith('{{~ _')) {
+      // Remove the special characters from the partial files
+      final formattedName = filePath
+          .replaceAll('{', '')
+          .replaceAll('}', '')
+          .replaceAll('~ ', '')
+          .replaceAll('.', '')
+          .trim();
+      newVars.putIfAbsent(formattedName, () => String.fromCharCodes(content));
+    }
+  });
+  return Future.value(newVars);
```

Comment:
```suggestion
  Map<String, dynamic> vars = {};
  final generator = await MasonGenerator.fromBundle(permissionHandlerBundle);
  generator.partials.forEach((filePath, content) {
    if (filePath.startsWith('{{~ _')) {
      // Remove the special characters from the partial files
      final formattedName = filePath
          .replaceAll('{', '')
          .replaceAll('}', '')
          .replaceAll('~ ', '')
          .replaceAll('.', '')
          .trim();
      vars.putIfAbsent(formattedName, () => String.fromCharCodes(content));
    }
  });
  return Future.value(vars);
```

Reply:
@luongvo resolved in [b41d394](https://github.com/nimblehq/flutter-templates/pull/203/commits/b41d39478c5a95ec11cc7f09f6966bf6a814f548)